pkgs/tools/graphics/astc-encoder/default.nix

```
@@ -0,0 +1,46 @@
+{ lib, stdenv, fetchFromGitHub, cmake }:
+
+stdenv.mkDerivation rec {
+  pname = "astc-encoder";
+  version = "2.5";
+  outputs = [ "out" ];
+
+  src = fetchFromGitHub {
+    owner = "ARM-software";
+    repo = "astc-encoder";
+    rev = "${version}";
+    sha256 = "0ff5jh40w942dz7hmgvznmpa9yhr1j4i9qqj5wy6icm2jb9j4pak";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  # Built with CMake flags from https://github.com/ARM-software/astc-encoder/blob/main/Docs/Building.md
+  # NEEDHELP: set the correct flags for the target architecture (arm etc.)
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DISA_AVX2=ON" "-DISA_SSE41=ON" "-DISA_SSE2=ON"
```

```suggestion
    "-DISA_AVX2=ON"
    "-DISA_SSE41=ON"
    "-DISA_SSE2=ON"
```

avx can be detected with ``enableAVX2 ? stdenv.hostPlatform.avx2Support``

---

Ahh I just noticed the comment about detecting AVX2 support. I think I can use that to select the correct build flag for the user's machine